Flujos de Control

1) Crear una variable que contenga un elemento del conjunto de números enteros y luego imprimir por pantalla si es mayor o menor a cero

In [ ]:
x = 5
if x > 0:
   print(str(x) + " es mayor que 0")
elif x < 0:
    print(str(x) + " es menor que 0")
else:
    print(str(x) + " es igual a 0")
    

2) Crear dos variables y un condicional que informe si son del mismo tipo de dato

In [ ]:
var1 = str(5)
var2 = '5'
if type(var1) == type(var2):
    print("var1 y var2 son del mismo tipo")
else:
    print("var 1 y var2 no son del mismo tipo")

3) Para los valores enteros del 1 al 20, imprimir por pantalla si es par o impar

In [ ]:
for num in range(1,21):
    if num%2 == 0:
        print(str(num) + " es número par")
    else:
        print(str(num) + " es número impar")
        


4) En un ciclo for mostrar para los valores entre 0 y 5 el resultado de elevarlo a la potencia igual a 3

In [ ]:
for num in range(6):
    print(str(num) + " elevado a la potencia 3 = " + str(num**3))

5) Crear una variable que contenga un número entero y realizar un ciclo for la misma cantidad de ciclos

In [ ]:
num=10
print("Realizaremos " + str(num) + " ciclos")
for i in range(1, num+1):
    print("ciclo " + str(i))


6) Utilizar un ciclo while para realizar el factorial de un número guardado en una variable, sólo si la variable contiene un número entero mayor a 0

In [ ]:
num = 15
numFactorial = 1
i = num
if i > 0:
    while i > 1:
        numFactorial *=i
        i-=1
    print("El factorial de " +str(num) + " es " + str(numFactorial))
else:
    print("Se debe ingresar un número mayor que 0")

7) Crear un ciclo for dentro de un ciclo while

In [ ]:
# Tablas de multiplicar del 1 al 5
tabla = 1
while tabla <= 5:
    for i in range(1, 11):
        print(str(tabla) + " x " + str(i) + " = " + str(tabla*i))
    tabla+=1
    print()
print("He terminado.....")    
        

8) Crear un ciclo while dentro de un ciclo for

In [ ]:
# Tablas de multiplicar del 1 al 5
j = 1
for tabla in range(1, 6):
    while j <= 10:
        print(str(tabla) + " x " + str(j) + " = " + str(tabla*j))
        j+=1
    print()
    j = 1
print("He terminado.....")   

9) Imprimir los números primos existentes entre 0 y 30

In [52]:

## Números primos con límite, considero que el número 1 no es primo, pues tiene solo un divisor.

tope = 300
for i in range(2, tope + 1):
    cont = 0
    for j in range(1, i+1):
        ciclo += 1
        if i % j == 0:
            cont += 1
    if cont <= 2:
        
        print(str(i) + " es número primo")


2 es número primo
3 es número primo
5 es número primo
7 es número primo
11 es número primo
13 es número primo
17 es número primo
19 es número primo
23 es número primo
29 es número primo
31 es número primo
37 es número primo
41 es número primo
43 es número primo
47 es número primo
53 es número primo
59 es número primo
61 es número primo
67 es número primo
71 es número primo
73 es número primo
79 es número primo
83 es número primo
89 es número primo
97 es número primo
101 es número primo
103 es número primo
107 es número primo
109 es número primo
113 es número primo
127 es número primo
131 es número primo
137 es número primo
139 es número primo
149 es número primo
151 es número primo
157 es número primo
163 es número primo
167 es número primo
173 es número primo
179 es número primo
181 es número primo
191 es número primo
193 es número primo
197 es número primo
199 es número primo
211 es número primo
223 es número primo
227 es número primo
229 es número primo
233 es número primo
239 es nú

10) ¿Se puede mejorar el proceso del punto 9? Utilizar las sentencias break y/ó continue para tal fin

In [53]:
tope = 300
ciclo1 = 0
for i in range(2, tope + 1):
    cont = 0
    for j in range(1, i+1):
        ciclo1 += 1
        if i % j == 0:
            cont += 1
        if cont > 2:
            break
    if cont <= 2:
        
        print(str(i) + " es número primo")
print(str(ciclo1) + " ciclos" )


2 es número primo
3 es número primo
5 es número primo
7 es número primo
11 es número primo
13 es número primo
17 es número primo
19 es número primo
23 es número primo
29 es número primo
31 es número primo
37 es número primo
41 es número primo
43 es número primo
47 es número primo
53 es número primo
59 es número primo
61 es número primo
67 es número primo
71 es número primo
73 es número primo
79 es número primo
83 es número primo
89 es número primo
97 es número primo
101 es número primo
103 es número primo
107 es número primo
109 es número primo
113 es número primo
127 es número primo
131 es número primo
137 es número primo
139 es número primo
149 es número primo
151 es número primo
157 es número primo
163 es número primo
167 es número primo
173 es número primo
179 es número primo
181 es número primo
191 es número primo
193 es número primo
197 es número primo
199 es número primo
211 es número primo
223 es número primo
227 es número primo
229 es número primo
233 es número primo
239 es nú

11) En los puntos 9 y 10, se diseño un código que encuentra números primos y además se lo optimizó. ¿Es posible saber en qué medida se optimizó?

In [54]:
print("Proceso sin optimizar: " + str(ciclo) + " ciclos")
print("Proceso optimizado: " + str(ciclo1) + " ciclos")
total = round((ciclo1 * 100)/ciclo, 2)
print("La optimización corresponde a un " + str(total) + "%" )


Proceso sin optimizar: 45613 ciclos
Proceso optimizado: 13726 ciclos
La optimización corresponde a un 30.09%


12) Si la cantidad de números que se evalúa es mayor a treinta, esa optimización crece?

In [55]:
print("Proceso sin optimizar: " + str(ciclo) + " ciclos")
print("Proceso optimizado: " + str(ciclo1) + " ciclos")
total = round((ciclo1 * 100)/ciclo, 2)
print("La optimización corresponde a un " + str(total) + "%" )

Proceso sin optimizar: 45613 ciclos
Proceso optimizado: 13726 ciclos
La optimización corresponde a un 30.09%


Aumentando la cantidad de números primos a obtener(300), vemos que la optimización disminuye a un 30.09%

13) Aplicando continue, armar un ciclo while que solo imprima los valores divisibles por 12, dentro del rango de números de 100 a 300

In [67]:
num=99
while num>=99 and num<=300:
    num += 1
    if num % 12 == 0:
        print(num)
    else:
        continue

108
120
132
144
156
168
180
192
204
216
228
240
252
264
276
288
300


14) Utilizar la función **input()** que permite hacer ingresos por teclado, para encontrar números primos y dar la opción al usario de buscar el siguiente

In [4]:
buscarprimo = True
while (buscarprimo):
    i = int(input("Digite el número del que quiere averiguar si es primo o no"))
    cont = 0
    for j in range(1, i+1):
        if i % j == 0:
            cont += 1
        if cont > 2:
            break
    if cont <= 2:
        print(str(i) + " es número primo")
    else:
        print(str(i) + " no es número primo")
    respuesta = int(input("Desea buscar otro número(si(1), no(0))"))
    if respuesta == 0:
        buscarprimo = False
    else:
        buscarprimo = True
    
print("Adios")

29 es número primo
173 es número primo
Adios


15) Crear un ciclo while que encuentre dentro del rango de 100 a 300 el primer número divisible por 3 y además múltiplo de 6

In [5]:
num=99
while num>=99 and num<=300:
    num += 1
    if num % 3 == 0 and num % 6 == 0:
        print(num)
        break
    else:
        continue

102
